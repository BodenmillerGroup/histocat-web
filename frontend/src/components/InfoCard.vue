<template>
  <v-card tile flat class="card overflow-y-auto">
    <v-card-title>Info</v-card-title>
    <v-divider />
    <v-card-title v-if="imageUrl">
      <v-img :src="imageUrl" class="grey lighten-2" width="380px">
        <template v-slot:placeholder>
          <v-row class="fill-height ma-0" align="center" justify="center">
            <v-progress-circular indeterminate color="grey lighten-5" />
          </v-row>
        </template>
      </v-img>
    </v-card-title>
    <v-list dense>
      <template v-for="item in items">
        <v-list-item dense inactive :key="item.name">
          <v-list-item-content>
            <v-list-item-title v-text="item.name" />
          </v-list-item-content>
          <v-list-item-action>
            <v-list-item-action-text v-text="item.value" />
          </v-list-item-action>
        </v-list-item>
      </template>
    </v-list>
  </v-card>
</template>

<script lang="ts">
import { apiUrl } from "@/env";
import { Component, Prop, Vue } from "vue-property-decorator";

@Component
export default class InfoCard extends Vue {
  @Prop(Object) readonly node;

  get imageUrl() {
    switch (this.node.item.type) {
      case "slide":
        return `${apiUrl}/slides/${this.node.item.id}/image`;
      case "panorama":
        return `${apiUrl}/panoramas/${this.node.item.id}/image`;
      default:
        return undefined;
    }
  }

  get items() {
    if (this.node.item.meta) {
      const items = Object.entries(this.node.item.meta);
      return items.map((item) => {
        return {
          name: item[0],
          value: item[1],
        };
      });
    } else {
      return [];
    }
  }
}
</script>

<style scoped>
.card {
  height: 40vh;
}
</style>
