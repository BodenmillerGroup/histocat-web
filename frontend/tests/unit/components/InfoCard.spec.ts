import { mount } from "@vue/test-utils";
import InfoCard from "@/components/InfoCard.vue";

import Vue from "vue";
import Vuetify from "vuetify";
Vue.use(Vuetify);

describe("InfoCard.vue", () => {
  it("renders correctly", () => {
    const wrapper = mount(InfoCard, {
      propsData: {
        node: {
          item: {
            type: "slide",
          },
        },
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
