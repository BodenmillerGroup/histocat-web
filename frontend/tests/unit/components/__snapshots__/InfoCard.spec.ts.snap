// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard.vue renders correctly 1`] = `
<div
  class="card overflow-y-auto v-card v-card--flat v-sheet theme--light rounded-0"
>
  <div
    class="v-card__title"
  >
    Info
  </div>
   
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
   
  <div
    class="v-card__title"
  >
    <div
      class="v-image v-responsive grey lighten-2 theme--light"
      style="width: 380px;"
    >
      <transition-stub
        mode="in-out"
        name="fade-transition"
      >
        <div
          class="v-image__image v-image__image--preload v-image__image--cover"
          style="background-position: center center;"
        />
      </transition-stub>
      <transition-stub
        appear="true"
        name="fade-transition"
      >
        <div
          class="v-image__placeholder"
        >
          <div
            class="row fill-height ma-0 align-center justify-center"
          >
            <div
              aria-valuemax="100"
              aria-valuemin="0"
              class="v-progress-circular v-progress-circular--visible v-progress-circular--indeterminate grey--text text--lighten-5"
              role="progressbar"
              style="height: 32px; width: 32px;"
            >
              <svg
                style="transform: rotate(0deg);"
                viewBox="22.857142857142858 22.857142857142858 45.714285714285715 45.714285714285715"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  class="v-progress-circular__overlay"
                  cx="45.714285714285715"
                  cy="45.714285714285715"
                  fill="transparent"
                  r="20"
                  stroke-dasharray="125.664"
                  stroke-dashoffset="125.66370614359172px"
                  stroke-width="5.714285714285714"
                />
              </svg>
              <div
                class="v-progress-circular__info"
              />
            </div>
          </div>
        </div>
      </transition-stub>
      <div
        class="v-responsive__content"
      />
    </div>
  </div>
   
  <div
    class="v-list v-sheet theme--light v-list--dense"
    role="list"
  />
</div>
`;
